src/main/java/com/google/api/codegen/config/FlatteningConfig.java

```diff
@@ -124,9 +124,9 @@
     // flattening.
     Map<String, FlatteningConfig> flatteningConfigs = new LinkedHashMap<>();
 
-    flatteningConfigs.putAll(flatteningConfigsFromGapicConfig);
-    // Let flattenings from proto annotations override flattenings from GAPIC config.
+    // Let flattenings from GAPIC config override flattenings from proto annotations.
```

Comment:
Here we are merging the configs from proto annotations and gapic config. But in other places in this PR we are using the new enum to pick one or the other, right? Or is that not the correct way to think about this?

Reply:
There's a TODO in [ProtoParser.java](https://github.com/googleapis/gapic-generator/blob/1add5a8ba75e29a80613fb2a187a46c08887434a/src/main/java/com/google/api/codegen/util/ProtoParser.java#L56) to toggle on and off parsing the proto annotations. 

After that is implemented, then `createFlatteningConfigsFromProtoFile()` should return an empty list because the ProtoParser was blind to any MethodSignature annotations in the proto file.

Reply:
I guess what I am saying is - it isn't clear to me why in some places we have threaded in the GapicConfigPresence enum value, while here we have not. If the reason is because we can be sure that createFlatteningConfigsFromProtoFile will return empty when GapicConfigPresence==PROVIDED, then can we apply that logic elsewhere and not thread the new enum through method signatures? We already have the protoParser available I think.

Reply:
Ok, removed the GapicConfigPresence!

Reply:
Ok, removed the GapicConfigPresence!